From 715964d7b404e5d4edf9220d37611efc3595ad65 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 9 Oct 2023 16:47:41 +0800
Subject: [PATCH] [riscv][compres-ptr] Fix LoadFeedback segment fault

Change-Id: Id07305855ea1bda68113903003a15aa4ca537733
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4921691
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#90316}
---
 src/codegen/riscv/macro-assembler-riscv.cc | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/codegen/riscv/macro-assembler-riscv.cc b/src/codegen/riscv/macro-assembler-riscv.cc
index b82ceb8b555..4ab069f9d23 100644
--- a/src/codegen/riscv/macro-assembler-riscv.cc
+++ b/src/codegen/riscv/macro-assembler-riscv.cc
@@ -6804,11 +6804,12 @@ void MacroAssembler::LoadFeedbackVector(Register dst, Register closure,
                                         Register scratch, Label* fbv_undef) {
   Label done;
   // Load the feedback vector from the closure.
-  LoadWord(dst, FieldMemOperand(closure, JSFunction::kFeedbackCellOffset));
-  LoadWord(dst, FieldMemOperand(dst, FeedbackCell::kValueOffset));
+  LoadTaggedField(dst,
+                  FieldMemOperand(closure, JSFunction::kFeedbackCellOffset));
+  LoadTaggedField(dst, FieldMemOperand(dst, FeedbackCell::kValueOffset));
 
   // Check if feedback vector is valid.
-  LoadWord(scratch, FieldMemOperand(dst, HeapObject::kMapOffset));
+  LoadTaggedField(scratch, FieldMemOperand(dst, HeapObject::kMapOffset));
   Lhu(scratch, FieldMemOperand(scratch, Map::kInstanceTypeOffset));
   Branch(&done, eq, scratch, Operand(FEEDBACK_VECTOR_TYPE));
 
-- 
2.35.1

